js/bundle/src/encoder.ts

```diff
@@ -278,3 +280,42 @@ function validateExchangeURL(urlString: string): void {
 function byteString(s: string): Uint8Array {
   return new TextEncoder().encode(s);
 }
+
+// Type guard for checking that the headers are in valid format.
+export function isHeaders(obj: any): Headers {
```

Comment:
I decided against using type predicates because there I'm forced into a boolean value that I dont want, but I can change it into a type predicate if that's the preferred way to handle these.